ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/auth/OAuthServiceClient.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.api.auth;
+
+import javax.inject.Inject;
+import org.eclipse.che.api.auth.shared.dto.OAuthToken;
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.AsyncRequestFactory;
+import org.eclipse.che.ide.rest.DtoUnmarshallerFactory;
+
+/**
+ * Serves connections to OauthAuthentication service. Allows to get OAuth tokens via callback or as
+ * promise.
+ *
+ * @author Sergii Leschenko
```

Comment:
Are you sure that I'm an author of this class?

Reply:
At least, that is what was written in [original](https://github.com/eclipse/che/blob/master/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/auth/OAuthServiceClientImpl.java) file :)